File: code/jvm/domain/src/main/java/pt/isel/domain/messages/Message.java
package pt.isel.domain.messages;

import pt.isel.domain.channels.Channel;
import pt.isel.domain.users.UserInfo;

import java.time.LocalDateTime;

public record Message(
        Long id,
        String content,
        UserInfo user,
        Channel channel,
        LocalDateTime createdAt
) {
}

File: code/jvm/domain/src/main/kotlin/pt/isel/domain/messages/Message.kt
package pt.isel.domain.messages

import pt.isel.domain.channels.Channel
import pt.isel.domain.users.UserInfo
import java.time.LocalDateTime

data class Message(
    val id: Long,
    val content: String,
    val user: UserInfo,
    val channel: Channel,
    val createdAt: LocalDateTime = LocalDateTime.now(),
)
